src/folly/FollyKeepaliveTimer.h

```diff
@@ -23,10 +23,13 @@ class FollyKeepaliveTimer : public KeepaliveTimer {
 
   void start(const std::shared_ptr<ConnectionAutomaton>& connection) override;
 
+  void keepaliveReceived() override;
+
  private:
   std::shared_ptr<ConnectionAutomaton> connection_;
   folly::EventBase& eventBase_;
   std::shared_ptr<bool> running_;
   std::chrono::milliseconds period_;
+  bool pending_{false};
```

Comment:
I suggest making this boolean member an atomic<bool> because you can potentially call it from a different thread, or add DCHECK in keepaliveReceived that it is executed on the same eventBase_.
This is only for cleanness.